packages/binding-mqtt/src/mqtt-broker-server.ts

```diff
@@ -410,11 +417,19 @@ export default class MqttBrokerServer implements ProtocolServer {
         });
     }
 
-    public stop(): Promise<void> {
+    public async stop(): Promise<void> {
         return new Promise<void>((resolve, reject) => {
-            if (this.broker === undefined) resolve();
+            if (this.broker !== undefined) {
+                this.broker.unsubscribe("*");
+                this.broker.end(true);
+            }
+
+            if (this.hostedBroker !== undefined) {
+                this.hostedServer.close();
+                this.hostedBroker.close(() => resolve());
+            }
 
-            this.broker.unsubscribe("*");
+            resolve();
         });
```

Comment:
As this function is labelled as `async`, is explicitly returning a Promise still necessary?

Reply:
Fixed using just two promises.